internal/connector/internal/handlers/connector_cluster.go

```diff
@@ -244,22 +245,22 @@ func (o *ConnectorClusterHandler) buildAddonParams(serviceAccount *api.ServiceAc
 		},
 		{
 			Id:    "client-id",
-			Value: serviceAccount.ClientID,
+			Value: cluster.ClientId,
 		},
 		{
 			Id:    "client-secret",
-			Value: serviceAccount.ClientSecret,
+			Value: cluster.ClientSecret,
 		},
 	}
 	return p
 }
 
-func (o *ConnectorClusterHandler) buildTokenURL(serviceAccount *api.ServiceAccount) (string, error) {
+func (o *ConnectorClusterHandler) buildTokenURL(cluster dbapi.ConnectorCluster) (string, error) {
 	u, err := url.Parse(o.KeycloakConfig.KafkaRealm.TokenEndpointURI)
```

Comment:
`KeycloakConfig.KafkaRealm.TokenEndpointURI`should be set on the basis of SelectSSOProvider
* if MAS_SSO then `KeycloakConfig.KafkaRealm.TokenEndpointURI`
* if REDHAT_SSO then `KeycloakConfig.RedhatSSORealm.TokenEndpointUR`

or use this which will return the set sso provider
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/services/sso/keycloak.go#L26

Reply:
Something like: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/894